lib/api_repo.js

```diff
@@ -231,14 +231,24 @@ ApiRepo.prototype.buildPackages =
                 cleanName.replace(new RegExp('-', 'g'), '/');
             opts.packageInfo.api.name = that.pkgPrefix + cleanName;
             opts.packageInfo.api.version = version;
-            opts.packageInfo.api.buildCommonProtos =
-                that.opts.buildCommonProtos;
+            if (version) {
+              opts.packageInfo.api.fullName =
+                  opts.packageInfo.api.name + '-' + version;
+            } else {
+              opts.packageInfo.api.fullName = opts.packageInfo.api.name;
+            }
             var semver = opts.packageInfo.api.semver[l];
             if (semver) {
               /* eslint-disable camelcase */
               opts.packageInfo.api.semantic_version = semver;
               /* eslint-enable camelcase */
             }
+            if (that.opts.protoGenPackageDeps) {
+              opts.packageInfo.api.protoGenPackageDeps =
+                  that.opts.protoGenPackageDeps.map(function(dep) {
+                    return that.pkgPrefix + dep;
```

Comment:
A TODO seems appropriate, because only the java alt template actually reads this field so far.